core/src/main/java/com/linecorp/armeria/common/logging/ContentPreviewerFactoryBuilder.java

```diff
@@ -0,0 +1,216 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.Charset;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.MediaTypeSet;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.logging.PreviewSpec.PreviewMode;
+import com.linecorp.armeria.internal.common.ArmeriaHttpUtil;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+
+/**
+ * A builder which builds a {@link ContentPreviewerFactory}.
+ */
+public final class ContentPreviewerFactoryBuilder {
+
+    //TODO(minwoox): Add setters for the seprate request and response previewer.
+
+    private final Builder<PreviewSpec> previewSpecsBuilder = ImmutableList.builder();
+    private int maxLength;
+    private Charset defaultCharset = ArmeriaHttpUtil.HTTP_DEFAULT_CONTENT_CHARSET;
+
+    ContentPreviewerFactoryBuilder() {}
+
+    /**
+     * Sets the maximum length of the produced preview.
+     */
+    public ContentPreviewerFactoryBuilder maxLength(int maxLength) {
+        checkArgument(maxLength > 0, "maxLength : %d (expected: > 0)", maxLength);
+        this.maxLength = maxLength;
+        return this;
+    }
+
+    /**
+     * Sets the default {@link Charset} used to produce the text preview when a charset is not specified in the
+     * {@code "content-type"} header.
+     *
+     * <p>Note that this charset is only for the text preview, not for the binary preview.</p>
+     */
+    public ContentPreviewerFactoryBuilder defaultCharset(Charset defaultCharset) {
+        this.defaultCharset = requireNonNull(defaultCharset, "defaultCharset");
+        return this;
+    }
+
+    /**
+     * Sets the specified {@link MediaTypeSet} to produce the text preview when the content type of the
+     * {@link RequestHeaders} or {@link ResponseHeaders} {@linkplain MediaTypeSet#match(MediaType) matches}
+     * the {@link MediaTypeSet}.
+     */
+    public ContentPreviewerFactoryBuilder text(MediaTypeSet mediaTypeSet) {
```

Comment:
How about accepting `MediaType...` and `Iterable<MediaType>`? `MediaTypeSet` is `Iterable<MediaType>`, so the two methods will cover `MediaTypeSet` as well.

Reply:
I was a little bit worried because it will make this builder have as twice methods.
But it seems not a big deal so let me make a change for that. Thanks!